dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -184,21 +209,31 @@ void updateRule(String id, RuleRequest rule) throws RuleNotFoundException, IOExc
     rulesTable.upsert(updatedRuleFields);
   }
 
-  Map<String, Object> retrieveRule(String id) throws RuleNotFoundException, IOException {
+  Map<String, Object> retrieveRule(NamespacedId namespacedId) throws RuleNotFoundException, IOException {
+    String ruleId = namespacedId.getId();
+    Namespace namespace = namespacedId.getNamespace();
+    String namespaceName = namespace.getName();
+    long namespaceGeneration = namespace.getGeneration();
+
     Collection<Field<?>> keyFields = new ArrayList<>();
-    keyFields.add(Fields.stringField(ID_COL, id));
+    keyFields.add(Fields.stringField(NAMESPACE_COL, namespaceName));
+    keyFields.add(Fields.longField(GENERATION_COL, namespaceGeneration));
+    keyFields.add(Fields.stringField(ID_COL, ruleId));
 
     Optional<StructuredRow> optionalStructuredRow = rulesTable.read(keyFields);
 
     if (!optionalStructuredRow.isPresent()) {
       throw new RuleNotFoundException(
-        String.format("Rule '%s' does not exist. Create it first with a POST request.", id)
+        String
+          .format("Rule '%s' in namespace '%s' does not exist. Create it first with a POST request.", ruleId, namespace)
```

Comment:
nit: keep the method call on the same line:

```
String.format("...",
              ruleId, namespace);
```

Reply:
Done.